static/js/stream_list.js

```diff
@@ -719,13 +726,25 @@ function maybe_select_stream (e) {
     }
 }
 
+function toggle_filter_displayed() {
+    if (0 === $('.stream-list-filter.notdisplayed').length) {
+        exports.clear_and_hide_search();
+    } else {
+        exports.initiate_search();
+    }
+}
+
 $(function () {
     $(".stream-list-filter").expectOne()
         .on('click', focus_stream_filter)
         .on('input', update_streams_for_search)
         .on('keydown', maybe_select_stream);
 });
 
+$(function () {
+    $("#streams_header").expectOne()
```

Comment:
This matches `#streams_inline_cog` as well, which may not be desired behaviour?

The easiest alternative is `#streams_handler > .sidebar-title` - but that wouldn't fire unless the word itself were clicked. I prefer this.